### PR TITLE
Sphinx depencies for 1.7.2

### PR DESCRIPTION
**Pull request by dhomeier**
Additional packages and updates required for a `Sphinx`-update (see also SF package submission [#4723](https://sourceforge.net/p/fink/package-submissions/4723/)), now to upstream `1.7.2`.
Provides dependencies not included in #82 - the Python3.6 variant of `html5lib-py` and update to `1.0.1` + a new dependency, and the `packaging` module required for all `Sphinx` tests to run successfully.

### CODE REVIEW

File: 10.9-libcxx/stable/main/finkinfo/libs/pythonmods/webencodings-py.info [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+Source: https://pypi.io/packages/source/w/webencodings/webencodings-%v.tar.gz
+Source-MD5: 32f6e261d52e57bf7e1c4d41546d15b8
+
+Depends: python%type_pkg[python]-shlibs
+BuildDepends: python%type_pkg[python], setuptools-tng-py%type_pkg[python]
+
+Description: Character encoding aliases for legacy pages
+
+DescDetail: <<
+This is a Python implementation of the WHATWG Encoding standard.
```
Comment (nieder) [inline 10.9-libcxx/stable/main/finkinfo/libs/pythonmods/webencodings-py.info]:
```
Validating package file /sw/fink/dists/local/distributions/10.9-libcxx/stable/main/finkinfo/libs/pythonmods/webencodings-py.info...
Warning: "descdetail" contains line(s) exceeding 79 characters. (webencodings-py.info)
```

File: 10.9-libcxx/stable/main/finkinfo/libs/pythonmods/packaging-py.info [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+CompileScript: <<
+  python%type_raw[python] setup.py build 
+<<
+
+InstallScript: <<
+  python%type_raw[python] setup.py install --root=%d
+<<
+
+InfoTest: <<
+  TestDepends: pytest-py%type_pkg[python], pretend-py%type_pkg[python]
```
Comment (nieder) [inline 10.9-libcxx/stable/main/finkinfo/libs/pythonmods/packaging-py.info]:
```
/sw/bin/python2.7 setup.py pytest || TESTFAIL=1
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: invalid command 'pytest'
```

Comment (nieder) [reply 10.9-libcxx/stable/main/finkinfo/libs/pythonmods/packaging-py.info]: `%p/bin/pytest-%type_raw[python] || exit 2` works for TestScript and you can then get rid of the TESTFAIL and .pyc lines.

File: 10.9-libcxx/stable/main/finkinfo/libs/pythonmods/webencodings-py.info (hunk 1 above)
Comment (dhomeier) [reply 10.9-libcxx/stable/main/finkinfo/libs/pythonmods/webencodings-py.info]: Should learn not to make "trivial" changes after building with '-m'...

File: 10.9-libcxx/stable/main/finkinfo/libs/pythonmods/packaging-py.info (hunk 2 above)
Comment (dhomeier) [reply 10.9-libcxx/stable/main/finkinfo/libs/pythonmods/packaging-py.info]: That's odd. What is your version of `setuptools-tng-py27`? With `32.3.0-1` I got
```
/sw/bin/python2.7 setup.py pytest || TESTFAIL=1
running pytest
running egg_info
writing requirements to packaging.egg-info/requires.txt
writing packaging.egg-info/PKG-INFO
writing top-level names to packaging.egg-info/top_level.txt
writing dependency_links to packaging.egg-info/dependency_links.txt
reading manifest file 'packaging.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no previously-included files found matching '.travis.yml'
warning: no previously-included files found matching 'dev-requirements.txt'
no previously-included directories found matching 'docs/_build'
no previously-included directories found matching 'tasks'
writing manifest file 'packaging.egg-info/SOURCES.txt'
running build_ext
===================================================================== test session starts ======================================================================
platform darwin -- Python 2.7.14, pytest-3.4.0, py-1.5.2, pluggy-0.6.0
rootdir: /scratch.noindex/fink.build/packaging-py27-17.1-1/packaging-17.1, inifile:
collected 29047 items                                                                                                                                          

tests/test_markers.py .......................ss......................................................................................................... [  0%]
...
```
`%p/bin/pytest-%type_raw[python]` works the same here, but I seem to remember some packages where calling it directly failed to collect all tests...

Comment (nieder) [reply 10.9-libcxx/stable/main/finkinfo/libs/pythonmods/packaging-py.info]: `i   setuptools-tng-py2  32.3.0-1`

Comment (dhomeier) [reply 10.9-libcxx/stable/main/finkinfo/libs/pythonmods/packaging-py.info]: OK, I know what to add to `TestDepends` should I need it...
```
i pytest-runner-py27   3.0-1   Add setup.py test support
```
Leaving this package as is now.

**Comment by nieder** [comment]
These look good to me. Pinging @danielj7 again because html5lib-py is his.

**Comment by nieder** [comment]
Whoops. I just checked in a different webencodings-py.info. I didn't realize it came from here. Feel free to overwrite mine with this one.

**Comment by dmacks** [comment]
@nieder you committed a slightly different webencodings-py.info as 27c88b3eeb1d04066eb0edca56340ceba6d18898, so this PR has a conflict. The already-committed one looks more in keeping with our current best practice of:
```
Depends: python%type_pkg[python]
```
rather than
```
Depends: python%type_pkg[python]-shlibs
BuildDepends: python%type_pkg[python]
```
and also includes the LICENSE file.

The committed one also has its TestScript ```|| exit 2``` whereas the PR is ```|| exit 1```. @dhomeier Are there  cases where we expect a failure and should ignore it? TestScript is unusual in that it wants rc=2 for actual failures to treat as fatal, whereas rc=1 is when you only want to warn but not treat as fatal.

I'm not sure the right git workflow here...will simply deleting this .info from the branch resolve the conflict, or does the current .info need to be copied into the branch?

**Comment by nieder** [comment]
To resolve the conflict by keeping the current file I committed, @dhomeier needs to make a commit in his branch that deletes his copy of webencodings-py.info. Alternatively, I don't know if copying the current .info in stable to his branch will also resolve the conflict.

**Comment by dmacks** [comment]
Fortunately the commits for the conflicting file (webencodings-py.info) were separate from the commits for the other files (html5lib-py.info and packaging-py.info) and there didn't seem to be any concerns about those other two, so I cherry-picked their commits. So now it's up to @dhomeier to decide if there are any advantages to the webencodings-py.info here and rebase on top of the current fink master, or else decide that we have there now is good enough and close this PR.

**Comment by dhomeier** [comment]
@nieder, I've used the conflict resolver on the page to move to your `webencodings-py.info` version and then pushed my additional updates to run the tests with a newer tox version, and the update for `tox-py` 3.0.0.
Despite setting `PYTHONPATH` to `%b/build/lib` I am still seeing tox downloading (or at least rebuilding) and installing its own copy of the module:
```
GLOB sdist-make: /scratch.noindex/fink.build/webencodings-py36-0.5.1-1/webencodings-0.5.1/setup.py
py36 create: /scratch.noindex/fink.build/webencodings-py36-0.5.1-1/webencodings-0.5.1/.tox/py36
py36 inst: /scratch.noindex/fink.build/webencodings-py36-0.5.1-1/webencodings-0.5.1/.tox/dist/webencodings-0.5.1.zip
py36 installed: webencodings==0.5.1
```
i.e. there is another copy installed in `%b/.tox/py36/lib/python3.6/site-packages/webencodings` - probably identical to the one built in `%b/build/lib`, but I've no idea how to make tox use the latter...

**Comment by nieder** [comment]
* tox-py is missing BDep on setuptools-scm
* `mocker` is provided by the package [pytest-mock](https://pypi.org/project/pytest-mock/). I quickly glanced at it, but it needs tox for its own tests :) Tox tests working are probably better than the pytest-mock tests if you want to avoid circular dependencies.
* Debian has this patch for skipping some tests when tox is not installed: https://salsa.debian.org/python-team/modules/tox/blob/master/debian/patches/build-time-test-adjustments.patch
It is for tox-2.5.0, don't know whether any of it will still work for 3.0.0.

**Comment by dhomeier** [comment]
The patch an be applied, but seems to address a different issue. Anyway the two script tests can be skipped the same way, as well as the stuff requiring `pytest-mock`, until that package is added to fink (strangely the `pytest` module seems to offer no way to test if a fixture is available).

**Comment by dhomeier** [comment]
I've been wondering, why not leaving the `InfoTest` in pytest-mock uncommented and build it without tests when required? But admittedly this can become messy, as seen in #139.
Also I was not able to run/load any pytest-mock tests at all before the package was installed, so making it loaf the package from `%b/build/lib` does not seem to work, neither with the version in master, nor with
`TOX_TESTENV_PASSENV="PYTHONPATH=%b/build/lib" %p/bin/tox-py%type_pkg[python] -e py%type_pkg[python]`
`fink -m rebuild` worked once the package was installed, but the `py3X` variants failed on one test:
```
running: /sw/bin/python3.6 /sw/lib/python3.6/site-packages/pytest.py --basetemp=/private/tmp/pytest-of-fink-bld/pytest-82/test_standalone_mock0/runpytest-0
     in: /private/tmp/pytest-of-fink-bld/pytest-82/test_standalone_mock0
============================= test session starts ==============================
platform darwin -- Python 3.6.5, pytest-3.5.1, py-1.5.3, pluggy-0.6.0
PyQt5 5.10.1 -- Qt runtime 5.7.1 -- Qt compiled 5.7.1
Matplotlib: 2.2.0
Freetype: 2.6.3
rootdir: /private/tmp/pytest-of-fink-bld/pytest-82/test_standalone_mock0, inifile: tox.ini
plugins: remotedata-0.2.0, qt-2.3.1, openfiles-0.2.0, mpl-0.9, mock-1.10.0, doctestplus-0.1.2, arraydiff-0.2, hypothesis-3.34.0
collected 1 item

test_standalone_mock.py .                                                [100%]

=========================== 1 passed in 0.06 seconds ===========================
============================================================= 1 failed, 47 passed in 4.22 seconds ==============================================================
Coverage.py warning: Module pytest_mock.py was never imported.
Coverage.py warning: No data was collected.
ERROR: InvocationError: '/sw/bin/coverage3.6 run --append --source=pytest_mock.py -m pytest test_pytest_mock.py'
___________________________________________________________________________ summary ____________________________________________________________________________
ERROR:   py36: commands failed
### execution of TOX_TESTENV_PASSENV="PYTHONPATH=/scratch.noindex/fink.build/pytest-mock-py36-1.10.0-1/pytest-mock-1.10.0/build/lib" failed, exit code 2
```
The two tests in tox-py are marked `skipif(os.environ.get('DEB_SKIP_TOX_TESTS'), ...)`, so the test suite itself completes.

**Comment by nieder** [comment]
The only conflict here seems to be updating tox-py.info to match what's in stable from when #199 was merged. I think it's good to go after that since the only remaining change is to webencondings-py.info

**Comment by nieder** [comment]
@dhomeier Do you actually see any tests being run for webencodings-py? I've tried running them via tox, pytest or tests.py directly, and all cases report 0 tests run.

This PR now conflicts with tox-py.info since that's now at 3.0.0 in git and I'm not sure webencodings tests are doing anything anyway. So I'm going to close this and specific changes in tox or webencodings should be done in different PRs. Reopen if needed.

**Pull request closed without merging**